projects/igniteui-angular/src/lib/calendar/calendar-base.ts

```diff
@@ -366,7 +366,12 @@ export class IgxCalendarBaseDirective implements ControlValueAccessor {
      * a single `Date` object.
      * Otherwise it is an array of `Date` objects.
      */
-    public set value(value: Date | Date[]) {
+    public set value(value: Date | Date[] | string) {
+        // Validate the date if it is of type string and it is IsoDate
+        if (typeof value === 'string') {
```

Comment:
This code is repeated 5 times, can it be extracted into a function or util file?